litr/src/main/java/com/linkedin/android/litr/render/GlThumbnailRenderer.kt

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2021 LinkedIn Corporation
+ * All Rights Reserved.
+ *
+ * Licensed under the BSD 2-Clause License (the "License").  See License in the project root for
+ * license information.
+ */
+package com.linkedin.android.litr.render
+
+import android.graphics.Bitmap
+import android.graphics.Paint
+import android.graphics.Point
+import android.graphics.SurfaceTexture
+import android.opengl.GLES20
+import android.opengl.Matrix
+import android.view.Surface
+import com.linkedin.android.litr.filter.GlFilter
+import com.linkedin.android.litr.filter.GlFrameRenderFilter
+import com.linkedin.android.litr.filter.video.gl.DefaultVideoFrameRenderFilter
+import java.nio.ByteBuffer
+import java.util.*
+
+
+/**
+ * A renderer that applies OpenGL filters to a bitmap, and returns a new bitmap.
+ */
+class GlThumbnailRenderer(filters: List<GlFilter>?) : ThumbnailRenderer {
```

Comment:
Note: In the future we should refactor this to use `GlVideoRenderer`. Two renderers behave differently, what this renderer produces is not always a true representation of end video.

Reply:
> Two renderers behave differently, what this renderer produces is not always a true representation of end video.

That's not ideal nor intended obviously. My rationale for having the separate renderer was based on the assumption that: a renderer is responsible only for providing the rendering resources (e.g. GL context, texture, vertex transformation, etc) needed for filters to draw to some target surface, and the actual rendering rules are defined in filters themselves. The two renderers didn't appear to have much in common in that respect. The main difference to consider was the way vertex transformation is defined -- but since the GlThumbnailRenderer always outputs the frame in the same aspect ratio, and because video is already rotated (AFAIK rotation is handled by the extractor for the respective media type), the matrix doesn't scale nor rotate

Reply:
I see your reasoning. I still think we can try to consolidate this in the future, but for now this is ok.